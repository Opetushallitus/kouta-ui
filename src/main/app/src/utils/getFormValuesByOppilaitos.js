import mapValues from 'lodash/mapValues';

import parseEditorState from './draft/parseEditorState';
import { isNumber } from './index';

const getFormValuesByOppilaitos = oppilaitos => {
  const {
    kielivalinta,
    tila,
    metadata: {
      osat,
      tietoaOpiskelusta,
      osoite: { osoite, postinumeroKoodiUri },
      esittely,
      wwwSivu,
      puhelinnumero,
      opiskelijoita,
      korkeakouluja,
      tiedekuntia,
      kampuksia,
      yksikoita,
      toimipisteita,
      akatemioita,
    },
  } = oppilaitos;

  return {
    kieliversiot: kielivalinta || [],
    tila,
    osat: osat || [],
    esittely: mapValues(esittely || {}, parseEditorState),
    yhteystiedot: {
      osoite: osoite || {},
      postinumero: postinumeroKoodiUri ? { value: postinumeroKoodiUri } : null,
      verkkosivu: wwwSivu || {},
      puhelinnumero: puhelinnumero || '',
    },
    tietoa: {
      osiot: (tietoaOpiskelusta || []).map(({ otsikkoKoodiUri }) => ({
        value: otsikkoKoodiUri,
      })),
      tiedot: (tietoaOpiskelusta || []).reduce(
        (acc, { otsikkoKoodiUri, teksti }) => {
          acc[otsikkoKoodiUri] = teksti || {};

          return acc;
        },
        {},
      ),
    },
    perustiedot: {
      opiskelijoita: isNumber(opiskelijoita) ? opiskelijoita : '',
      korkeakouluja: isNumber(korkeakouluja) ? korkeakouluja : '',
      tiedekuntia: isNumber(tiedekuntia) ? tiedekuntia : '',
      kampuksia: isNumber(kampuksia) ? kampuksia : '',
      yksikoita: isNumber(yksikoita) ? yksikoita : '',
      toimipisteita: isNumber(toimipisteita) ? toimipisteita : '',
      akatemioita: isNumber(akatemioita) ? akatemioita : '',
    },
  };
};

export default getFormValuesByOppilaitos;
