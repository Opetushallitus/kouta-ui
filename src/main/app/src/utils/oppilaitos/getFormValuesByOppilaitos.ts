import _ from 'lodash';

import { parseEditorState } from '#/src/components/Editor/utils';

export const getFormValuesByOppilaitos = oppilaitos => {
  const {
    kielivalinta,
    tila,
    esikatselu = false,
    logo,
    teemakuva,
    metadata: {
      tietoaOpiskelusta,
      yhteystiedot = [],
      hakijapalveluidenYhteystiedot: hy,
      esittely,
      opiskelijoita,
      korkeakouluja,
      tiedekuntia,
      kampuksia,
      yksikoita,
      toimipisteita,
      akatemioita,
      wwwSivu,
    },
  } = oppilaitos;

  return {
    kieliversiot: kielivalinta || [],
    tila,
    esikatselu,
    esittely: _.mapValues(esittely || {}, parseEditorState),
    yhteystiedot: yhteystiedot.map(yhteystieto => ({
      nimi: yhteystieto.nimi || {},
      postiosoite: yhteystieto.postiosoite?.osoite || {},
      postinumero: yhteystieto.postiosoite?.postinumeroKoodiUri
        ? { value: yhteystieto.postiosoite.postinumeroKoodiUri }
        : null,
      kayntiosoite: yhteystieto.kayntiosoite?.osoite || {},
      kayntiosoitePostinumero: yhteystieto.kayntiosoite?.postinumeroKoodiUri
        ? { value: yhteystieto.kayntiosoite.postinumeroKoodiUri }
        : null,
      puhelinnumero: yhteystieto.puhelinnumero || {},
      sahkoposti: yhteystieto.sahkoposti || {},
    })),
    hakijapalveluidenYhteystiedot: hy
      ? {
          nimi: hy.nimi || {},
          postiosoite: hy.postiosoite?.osoite || {},
          postinumero: hy.postiosoite?.postinumeroKoodiUri
            ? {
                value: hy.postiosoite.postinumeroKoodiUri,
              }
            : null,
          kayntiosoite: hy.kayntiosoite?.osoite || {},
          kayntiosoitePostinumero: hy.kayntiosoite?.postinumeroKoodiUri
            ? { value: hy.kayntiosoite.postinumeroKoodiUri }
            : null,
          puhelinnumero: hy.puhelinnumero || {},
          sahkoposti: hy.sahkoposti || {},
        }
      : null,
    tietoa: {
      osiot: (tietoaOpiskelusta || []).map(({ otsikkoKoodiUri }) => ({
        value: otsikkoKoodiUri,
      })),
      tiedot: (tietoaOpiskelusta || []).reduce(
        (acc, { otsikkoKoodiUri, teksti }) => {
          acc[otsikkoKoodiUri] = _.mapValues(teksti || {}, parseEditorState);

          return acc;
        },
        {}
      ),
    },
    perustiedot: {
      opiskelijoita: _.isNumber(opiskelijoita) ? opiskelijoita : '',
      korkeakouluja: _.isNumber(korkeakouluja) ? korkeakouluja : '',
      tiedekuntia: _.isNumber(tiedekuntia) ? tiedekuntia : '',
      kampuksia: _.isNumber(kampuksia) ? kampuksia : '',
      yksikoita: _.isNumber(yksikoita) ? yksikoita : '',
      toimipisteita: _.isNumber(toimipisteita) ? toimipisteita : '',
      akatemioita: _.isNumber(akatemioita) ? akatemioita : '',
      logo,
      wwwSivuUrl: wwwSivu?.url || {},
      wwwSivuNimi: wwwSivu?.nimi || {},
    },
    teemakuva,
  };
};
