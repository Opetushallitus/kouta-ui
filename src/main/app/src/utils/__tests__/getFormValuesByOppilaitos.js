import getFormValuesByOppilaitos from '../getFormValuesByOppilaitos';

test('getFormValuesByOppilaitos returns correct form values given oppilaitos', () => {
  const values = getFormValuesByOppilaitos({
    kielivalinta: ['fi', 'sv'],
    tila: 'tallennettu',
    metadata: {
      akatemioita: 1,
      esittely: {
        fi: '<p><em>Fi esittely</em></p>',
        sv: '<p><em>Sv esittely</em></p>',
      },
      kampuksia: 3,
      korkeakouluja: 5,
      opiskelijoita: 100,
      osat: ['1.1.1.1', '1.1.1.2'],
      osoite: {
        osoite: {
          fi: 'Fi osoite',
          sv: 'Sv osoite',
        },
        postinumeroKoodiUri: 'postinumero_1#1',
      },
      puhelinnumero: '123456',
      tiedekuntia: 4,
      tietoaOpiskelusta: [
        {
          otsikkoKoodiUri: 'osio_1#1',
          teksti: {
            fi: 'Fi tiedot',
            sv: 'Sv tiedot',
          },
        },
      ],
      toimipisteita: 9,
      wwwSivu: {
        fi: 'www.verkkosivu.fi',
        sv: 'www.verkkosivu.sv',
      },
      yksikoita: 2,
    },
  });

  expect(values).toMatchSnapshot();
});
