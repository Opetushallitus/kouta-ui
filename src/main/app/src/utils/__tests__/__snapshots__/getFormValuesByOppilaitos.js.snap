// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getFormValuesByOppilaitos returns correct form values given oppilaitos 1`] = `
Object {
  "esittely": Object {
    "fi":       "<p><em>Fi esittely</em></p>",
    "sv":       "<p><em>Sv esittely</em></p>",
  },
  "kieliversiot": Array [
    "fi",
    "sv",
  ],
  "osat": Array [
    "1.1.1.1",
    "1.1.1.2",
  ],
  "perustiedot": Object {
    "akatemioita": 1,
    "kampuksia": 3,
    "korkeakouluja": 5,
    "opiskelijoita": 100,
    "tiedekuntia": 4,
    "toimipisteita": 9,
    "yksikoita": 2,
  },
  "tietoa": Object {
    "osiot": Array [
      Object {
        "value": "osio_1#1",
      },
    ],
    "tiedot": Object {
      "osio_1#1": Object {
        "fi": "Fi tiedot",
        "sv": "Sv tiedot",
      },
    },
  },
  "tila": "tallennettu",
  "yhteystiedot": Object {
    "osoite": Object {
      "fi": "Fi osoite",
      "sv": "Sv osoite",
    },
    "postinumero": Object {
      "value": "postinumero_1#1",
    },
    "puhelinnumero": "123456",
    "verkkosivu": Object {
      "fi": "www.verkkosivu.fi",
      "sv": "www.verkkosivu.sv",
    },
  },
}
`;
